src/constructors/mycielskian.c

```diff
@@ -0,0 +1,119 @@
+/*
+   IGraph library.
+   Copyright (C) 2022  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_constructors.h"
+#include "igraph_conversion.h"
+
+#include "igraph_interface.h"
+
+#include "core/interruption.h"
+#include "math/safe_intop.h"
+
+igraph_error_t igraph_mycielskian(igraph_t *res, const igraph_t *graph, igraph_integer_t k) {
+    if (k < 0) {
+        IGRAPH_ERROR("Number of iterations (k) must be non-negative", IGRAPH_EINVAL);
+    }
+    if (graph == NULL || res == NULL) {
+        IGRAPH_ERROR("Invalid input/output graph", IGRAPH_EINVAL);
+    }
```

Comment:
```suggestion
```

We usually do not check for errors that can only be committed in C. The principle is to validate input that can be passed from high-level interfaces.

_If_ we added a check here, it should be an `IGRAPH_ASSERT` instead, but in this case let us not add any check.